library/algorithms/rbit.lisp

```diff
@@ -0,0 +1,82 @@
+;;;; rbit.lisp
+
+(defpackage #:coalton-library/algorithms/rbit
```

Comment:
can we make the following:

- `rbit-package.lisp` which just contains this package definition
- `rbit-sbcl-arm64.lisp` which contains the SBCL VOPs and the like
- `rbit-portable.lisp` which contains a portable (possibly slow) implementation of the same functions

Then all the code later can safety use `rbit`.

I also would like to choose a better name: `reverse-bits` or `bit-reverse` or something. You can decide. But not `RBIT` which is too close to the ARM64 name.